driver-core/src/main/com/mongodb/internal/binding/AsyncSingleConnectionReadBinding.java

```diff
@@ -70,6 +72,11 @@ public ServerApi getServerApi() {
         return serverApi;
     }
 
+    @Override
+    public RequestContext getRequestContext() {
+        return IgnorableRequestContext.INSTANCE;
```

Comment:
Should we just remove `AsyncSingleConnectionReadBinding` together with `AsyncSingleConnectionReadBindingSpecification` given that `AsyncSingleConnectionReadBinding` is not used anywhere else?